tests/mail/tests.py

```
@@ -1552,6 +1552,27 @@ def test_all_params_can_be_set_before_send(self):
         self.assertIs(email.get_connection(), new_connection)
         self.assertNotIn("original", message.as_string())
 
+    def test_unicode_attachment_encoding(self):
+        """
+        Test that attaching a message containing non-ASCII characters does not
+        raise a UnicodeEncodeError.
+        """
+        att = """\
+        Subject: attachment
+        Content-Type: text/plain; charset=utf-8
+        Content-Transfer-Encoding: 8bit
+
+        ¡8-bit content!
+        """.encode()
+
+        email = EmailMessage(to=["test@example.com"])
+        email.attach("attachment.eml", att, "message/rfc822")
+
+        try:
+            email.message().as_bytes()
+        except UnicodeEncodeError as e:
+            self.fail(f"UnicodeEncodeError was raised: {e}")
```

Suggestion: Remove the exception handling. That way the test case failure has all the exception data. There's no need to catch an error just to raise a failure - let the test fail

---

done